### PR TITLE
Security/replace deprecated npm sass

### PR DESCRIPTION
**Pull request by MCrawleyHomeOffice**
Removed npm-sass, added dart-sass, pulled in any dependent logic from npm-sass and fixed all linting issues.

### CODE REVIEW

File: tasks/sass/index.js [hunk 1]
```diff
@@ -17,9 +18,16 @@ module.exports = config => {
...
       const aliases = {};
       if (config.theme) {
+        console.log('Applying theme: ' + config.theme);
         aliases.$$theme = `hof-theme-${config.theme}`;
       }
-      sass(config.sass.src, { aliases }, (err, result) => err ? reject(err) : resolve(result.css));
+
+      sass.render({
+        file: config.sass.src,
+        importer: importer({ aliases })
```
Comment (Alex-Swann) [inline tasks/sass/index.js]: needs to be 
```
sass.render({
        file: config.sass.src,
        importer: importer({ aliases }),
        aliases
      }, function (err, result) {
```

Comment (MCrawleyHomeOffice) [reply tasks/sass/index.js]: Thanks @Alex-Swann . Are you able to shed some light on what this is actually doing? I can't find any documentation related to this.

Comment (MCrawleyHomeOffice) [reply tasks/sass/index.js]: Resolved.

File: helpers/resolver/nearest-package-root.js [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+  const dirnameOfImportOrigin = path.dirname(importOriginPath);
+  let handleFoundPath;
+
+  const promise = new Promise(function (resolve, reject) {
+    handleFoundPath = function (err, nearestPackageParent) {
+      if (err) {
+        try {
+          return resolve(path.dirname(require.resolve(`${packageName}/package.json`)));
+        } catch (e) {
+          console.log(`Failed to find module ${packageName}/package.json for ${importOriginPath}`, e);
```
Comment (Alex-Swann) [inline helpers/resolver/nearest-package-root.js:18]: nice

**Review by Alex-Swann — APPROVED** [review]

**Pull request merged**